test/integration/ctl/ctl_create_cr_test.go

```diff
@@ -0,0 +1,211 @@
+		if err != nil {
+			t.Fatal(err)
+		}
+	}
+	tests := map[string]struct {
+		inputFile      string
+		inputArgs      []string
+		inputNamespace string
+		keyFilename    string
+
+		expErr         bool
+		expNamespace   string
+		expName        string
+		expKeyFilename string
+	}{
+		"v1alpha2 Certificate given": {
+			inputFile:      testdataPath + "create_cr_cert_with_ns1.yaml",
+			inputArgs:      []string{cr1Name},
+			inputNamespace: ns1,
+			keyFilename:    "",
+			expErr:         false,
+			expNamespace:   ns1,
+			expName:        cr1Name,
+			expKeyFilename: cr1Name + ".key",
+		},
+		"v1alpha3 Certificate given": {
+			inputFile:      testdataPath + "create_cr_v1alpha3_cert_with_ns1.yaml",
+			inputArgs:      []string{cr2Name},
+			inputNamespace: ns1,
+			keyFilename:    "",
+			expErr:         false,
+			expNamespace:   ns1,
+			expName:        cr2Name,
+			expKeyFilename: cr2Name + ".key",
+		},
+		"conflicting namespaces defined in flag and file": {
+			inputFile:      testdataPath + "create_cr_cert_with_ns1.yaml",
+			inputArgs:      []string{cr3Name},
+			inputNamespace: ns2,
+			keyFilename:    "",
+			expErr:         true,
+			expNamespace:   "",
+			expName:        "",
+			expKeyFilename: "",
+		},
+		"file passed in defines resource other than certificate": {
+			inputFile:      testdataPath + "create_cr_issuer.yaml",
+			inputArgs:      []string{cr4Name},
+			inputNamespace: ns1,
+			keyFilename:    "",
+			expErr:         true,
+			expNamespace:   "",
+			expName:        "",
+			expKeyFilename: "",
+		},
+		"path to file to store private key provided": {
+			inputFile:      testdataPath + "create_cr_cert_with_ns1.yaml",
+			inputArgs:      []string{cr1Name},
+			inputNamespace: ns1,
+			keyFilename:    "test.key",
+			expErr:         false,
+			expNamespace:   ns1,
+			expName:        cr1Name,
+			expKeyFilename: "test.key",
+		},
+	}
+
+	for name, test := range tests {
+		// Run ctl create cr command with input options
+		t.Run(name, func(t *testing.T) {
+			streams, _, _, _ := genericclioptions.NewTestIOStreams()
+
+			// Options to run create CR command
+			opts := &certificaterequest.Options{
+				CMClient:         cmCl,
+				RESTConfig:       config,
+				IOStreams:        streams,
+				CmdNamespace:     test.inputNamespace,
+				EnforceNamespace: test.inputNamespace != "",
+				KeyFilename:      test.keyFilename,
+			}
+
+			opts.InputFilenames = []string{test.inputFile}
+
+			err := opts.Run(test.inputArgs)
+
+			if err != nil {
+				if !test.expErr {
+					t.Errorf("got unexpected error when trying to create CR: %v", err)
+				}
+				t.Logf("got an error, which was expected, details: %v", err)
+				return
+			} else {
+				// got no error
+				if test.expErr {
+					t.Errorf("expected but got no error when creating CR")
+				}
+			}
+
+			// Finished creating CR, check if everything is expected
+			crName := test.inputArgs[0]
```

Comment:
We don’t appear to test the case where the name is auto generated with this from what I can see? Perhaps we can list all CRs and assert that one of them exists and it’s name has the Certificates name as a prefix?